app/models/manageiq/providers/amazon/inventory/persister/shared/storage_collections.rb

```diff
@@ -0,0 +1,58 @@
+module ManageIQ::Providers::Amazon::Inventory::Persister::Shared::StorageCollections
+  extend ActiveSupport::Concern
+
+  included do
+    # Builder class for Storage
+    def storage
+      ::ManagerRefresh::InventoryCollection::Builder::StorageManager
+    end
+
+    def add_cloud_volumes(extra_properties = {})
+      add_collection(storage, :cloud_volumes, extra_properties) do |builder|
+        builder.add_properties(
+          :model_class => ::ManageIQ::Providers::Amazon::StorageManager::Ebs::CloudVolume
+        )
+
+        builder.add_builder_params(
+          :ems_id => ->(persister) { persister.manager.try(:ebs_storage_manager).try(:id) || persister.manager.id }
```

Comment:
this is all in persister context now? So we should be able to just do:
```
builder.add_builder_params(
  :ems_id => manager.try(:ebs_storage_manager).try(:id) || persister.manager.id
)
```

probably we could extract the managers bit into method
```ruby
def block_storage_manager
  persister.manager.try(:ebs_storage_manager).try(:id) || persister.manager.id
end

def object_storage_manager
  persister.manager.try(:s3_storage_manager).try(:id) || persister.manager.id
end
```

Reply:
Yes, it's possible, but I wanted to keep the same approach in core classes and the rest. 
New developer would be confused, what (s)he have to use on which place, this way there is no source of confusing and result is the same.

Reply:
Right, so probably lets just drop the lambda? That should work consistent across ```core classes and the rest``` right?

Reply:
I thought about keeping lambda...but you're right, it's not as difficult and nobody should be confused